File: semestr-6/Wstep do kryptologii/lab-2/cezar.go
package main

import (
	"os"
	"strconv"
)

func get_num(chr byte) int {
	return int(chr - 'a')
}

func get_char(num int) byte {
	return byte((num+26)%26 + 'a')
}

func transform(char byte, key int)

func main() {

	args := os.Args[1:]

	key, _ := strconv.Atoi(args[0])
	plaintext := args[1]

	cipher := ""

	for i := 0; i < len(plaintext); i++ {
		num := get_num(plaintext[i])
		if num != -1 {
			num = (num + key) % 26
			cipher += string(get_char(num))
		} else {
			cipher += string(plaintext[i])
		}
	}
	
	println(cipher)
}

File: semestr-6/Wstep do kryptologii/lab-2/cezar.py
# zaszyfruj słowo kryptologia
import sys

def get_num(char,k):
    n = ord(char)-ord('a')
    letter = '+' if k>=0 else ''
    print(f'{char} = {n} -> {n}{letter}{k} = {n+k} -> {n+k}%26 -> {(n+k)%26} = {get_char((n+k)%26)}')
    return ord(char) - ord('a')

def get_char(num):
    return chr((num+26)%26 + ord('a'))

def caesar_cipher(text, k):
    alphabet = 'abcdefghijklmnopqrstuvwxyz'
    return ''.join([get_char(get_num(c,k) + k) if c in alphabet else c for c in text])

k = int(sys.argv[1])
text = sys.argv[2]

print(caesar_cipher(text, k))